source/slang/core.meta.slang

```diff
@@ -1197,6 +1197,11 @@ for (auto op : binaryOps)
 }
 }}}}
 
+// Specialized function
+
+__intrinsic_op
+int getStringHash(String string);
```

Comment:
Rationale for `int` vs. `uint` here?

Reply:
Only that is what GetHashCode returns - int. This is probably not entirely desirable not least that it means the hash is target int width dependent. Making uint is probably better - if not just for dumping.

Reply:
I suspect that we'll want to introduce a specific `StringHash` type into the stdlib, and probably make it include explicit conversion to/from `uint` rather than `int` (just to avoid this detail of the `GetHashCode` design leaking into our public interface).